### PR TITLE
[Feature]implement Lambda array_filter()

### PR DESCRIPTION
**Pull request by fzhedu**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

array_filter(array, array_bool) use array_bool to filter array, where the array_bool item is not null or not zero. 
if array_bool.size() != array.size(), the result array is decided by the valid items within min(array_bool, array).
if the array_bool is NULL, it does filter all elements by design. 
if any of array or array_bool is const null, return null directly.

for example:
```
array=[11,2], array_bool=[0,1,null,1], res = [2].
array=[11,2], array_bool=[1], res = [11].
array=[11,2], array_bool=NULL, res = [].
array=null, array_bool[], res = NULL.
```

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: be/src/exprs/vectorized/array_functions.tpp [hunk 1]
```diff
@@ -1033,4 +1033,86 @@ private:
...
+            filter = down_cast<ArrayColumn*>(nullable_column->data_column().get());
+            null_map = nullable_column->null_column().get();
+        } else {
+            filter = down_cast<ArrayColumn*>(raw_filter.get());
+        }
+        for (size_t i = 0; i < src_column->size(); ++i) {
+            if (desc_null_map == nullptr || !desc_null_map->get_data()[i]) {
+                // null filter is regard as invalid, so does not filter items.
+                if (null_map != nullptr && null_map->get_data()[i]) {
+                    dest_column->append(*src_column, i, 1);
```
Comment (kangkaisen) [inline be/src/exprs/vectorized/array_functions.tpp:1095]: could we use append_selective method here? it performance should better than append method.

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/FunctionAnalyzer.java [hunk 2]
```diff
@@ -153,6 +153,12 @@ private static void analyzeBuiltinAggFunction(FunctionCallExpr functionCallExpr)
             }
         }
 
+        if (fnName.getFunction().equalsIgnoreCase(FunctionSet.ARRAY_FILTER)) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/FunctionAnalyzer.java:156]: please change all `equalsIgnoreCase` to `equals`  in analyzeBuiltinAggFunction method. thanks.

Comment (fzhedu) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/FunctionAnalyzer.java]: solved

File: be/src/exprs/vectorized/array_functions.tpp (hunk 1 above)
Comment (fzhedu) [reply be/src/exprs/vectorized/array_functions.tpp]: updated according to your ideas.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 51 / 53 (96.23%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/FunctionAnalyzer.java|33|35|94.29%|[157, 158]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|18|18|100.00%|[]|

**Review by kangkaisen — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/FunctionAnalyzer.java [hunk 3]
```diff
@@ -27,7 +27,7 @@ public static void analyze(FunctionCallExpr functionCallExpr) {
         }
 
         FunctionName fnName = functionCallExpr.getFnName();
-        if (fnName.getFunction().equalsIgnoreCase(FunctionSet.DATE_TRUNC)) {
+        if (fnName.getFunction().equals(FunctionSet.DATE_TRUNC)) {
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/FunctionAnalyzer.java:30]: why ?

**Review by murphyatwork — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by satanson — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/FunctionAnalyzer.java (hunk 3 above)
Comment (fzhedu) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/FunctionAnalyzer.java:30]: suggested by https://github.com/StarRocks/starrocks/pull/10323#discussion_r962578249

**Pull request merged**